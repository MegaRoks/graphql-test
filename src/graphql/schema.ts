import { buildSchema } from 'graphql';

export default buildSchema(`
    type BookType {
        bookId: Int!
        name: String!
        pageCount: Int!
        authorId: Int!
        author: AuthorType!
    }

    type AuthorType {
        authorId: Int!
        name: String!
    }

    type Query {
        getBooks: [BookType!]!
    }
`);
