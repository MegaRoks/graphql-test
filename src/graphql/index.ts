import { join } from 'path';
import { ApolloServer } from 'apollo-server';
import { buildSchema } from 'type-graphql';

export const getConnectionGraphQL = async () => {
    try {
        const schema = await buildSchema({ resolvers: [`${join(__dirname, '..', 'resolvers', '*.resolver.ts')}`] });
        return new ApolloServer({ schema });
    } catch (err) {
        console.error('err', err);
    }
};
