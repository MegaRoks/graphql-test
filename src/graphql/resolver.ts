import { Author } from './../entities/author.entity';
import connection from './../database';

const author = {
    authorId: 1,
    name: 'Author 1',
};

const books = [
    {
        bookId: 1,
        name: 'Book 1',
        pageCount: 22,
        authorId: 1,
        author,
    },
];

export default {
    async getBooks() {
        return connection
            .then((connect) => {
                const authorRepository = connect.getRepository(Author);
                return authorRepository.find();
            })
            .catch((err) => {
                console.error('error', err);
            });
    },
};
