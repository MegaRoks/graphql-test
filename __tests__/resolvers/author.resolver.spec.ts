import { makeExecutableSchema } from '@graphql-tools/schema';
import { addMocksToSchema } from '@graphql-tools/mock';
import { graphql } from 'graphql';

describe('Author Resolver Tests', () => {
    const schemaString = `
        type Author {
            authorId: String!
            name: String!
            createdAt: String!
            updatedAt: String!
        }

        type Query {
            getAuthorsList: [Author!]!
            getAuthor(authorId: String!): Author!
        }

        input AuthorInput {
            name: String!
        }

        type Mutation {
            createAuthor(data: AuthorInput!): Author!
            updateAuthor(authorId: String!, data: AuthorInput!): Author!
            deleteAuthor(authorId: String!): Boolean!
        }
    `;

    const schema = makeExecutableSchema({ typeDefs: schemaString });
    const schemaWithMocks = addMocksToSchema({ schema });

    it('should get author list', async () => {
        const query = `
            mutation {
                getAuthorsList {
                    authorId name createdAt updatedAt
                }
            }
        `;

        const response = await graphql(schemaWithMocks, query);

        expect(response).toMatchSnapshot();
    });

    it('should get author', async () => {
        const query = `
            query {
                getAuthor(authorId: "qwe123") {
                    authorId name createdAt updatedAt
                }
            }
        `;

        const response = await graphql(schemaWithMocks, query);

        expect(response).toMatchSnapshot();
    });

    it('should create author', async () => {
        const query = `
            mutation {
                createAuthor(data: { name: "Author 1"}) {
                    authorId name createdAt updatedAt
                }
            }
        `;

        const response = await graphql(schemaWithMocks, query);

        expect(response).toMatchSnapshot();
    });

    it('should update author', async () => {
        const query = `
            mutation {
                updateAuthor(authorId: "qwe123", data: { name: "Author 2"}) {
                    authorId name createdAt updatedAt
                }
            }
        `;

        const response = await graphql(schemaWithMocks, query);

        expect(response).toMatchSnapshot();
    });

    it('should delete author', async () => {
        const query = `
            mutation {
                deleteAuthor(authorId: "qwe123")
            }
        `;

        const response = await graphql(schemaWithMocks, query);

        expect(response).toMatchSnapshot();
    });
});
