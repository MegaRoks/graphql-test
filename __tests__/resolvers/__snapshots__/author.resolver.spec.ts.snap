// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Resolver Tests should create author 1`] = `
Object {
  "data": Object {
    "createAuthor": Object {
      "authorId": "Hello World",
      "createdAt": "Hello World",
      "name": "Hello World",
      "updatedAt": "Hello World",
    },
  },
}
`;

exports[`Author Resolver Tests should delete author 1`] = `
Object {
  "data": Object {
    "deleteAuthor": true,
  },
}
`;

exports[`Author Resolver Tests should get author 1`] = `
Object {
  "data": Object {
    "getAuthor": Object {
      "authorId": "Hello World",
      "createdAt": "Hello World",
      "name": "Hello World",
      "updatedAt": "Hello World",
    },
  },
}
`;

exports[`Author Resolver Tests should get author list 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Cannot query field "getAuthorsList" on type "Mutation".],
  ],
}
`;

exports[`Author Resolver Tests should update author 1`] = `
Object {
  "data": Object {
    "updateAuthor": Object {
      "authorId": "Hello World",
      "createdAt": "Hello World",
      "name": "Hello World",
      "updatedAt": "Hello World",
    },
  },
}
`;
